client-ui/src/app/components/time-frame-picker/time-frame-picker.component.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import * as moment from 'moment';
+import {NgbDateStruct, NgbTimeStruct} from "@ng-bootstrap/ng-bootstrap";
+import {DateRange, Timestamp} from "./time-frame-picker.model";
+
+type ViewModel = [LocalDateTimeModel, LocalDateTimeModel]
+
+interface LocalDateTimeModel {
+  time: NgbTimeStructWithMillis
+  date: NgbDateStruct
+}
+
+interface NgbTimeStructWithMillis extends NgbTimeStruct {
+  millis: number
+}
+
+@Component({
+  selector: 'time-frame-picker',
+  templateUrl: './time-frame-picker.component.html'
+})
+export class TimeFramePickerComponent {
+  model: ViewModel
+  modelBoundaries: ViewModel
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'modelBoundaries' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)